sonic_platform_base/module_base.py

```diff
@@ -21,6 +21,8 @@ class ModuleBase(device_base.DeviceBase):
     MODULE_TYPE_SUPERVISOR = "SUPERVISOR"
     MODULE_TYPE_LINE    = "LINE-CARD"
     MODULE_TYPE_FABRIC  = "FABRIC-CARD"
+    MODULE_TYPE_DPU  = "DPU"
+    MODULE_TYPE_SWITCH  = "SWITCH"
```

Comment:
Do we need to have module type "SWITCH"? What is its purpose?

Reply:
This is a Microsoft requirement.  My understanding is that, "SWITCH" is the functional  equivalent of "SUPERVISOR" of a modular chassis.

Reply:
@rameshraghupathy @oleksandrivantsiv one purpose is to have the output to display different modules in a chassis/smart switch

![image](https://github.com/sonic-net/sonic-platform-common/assets/45705344/6c25f562-0047-416c-bb09-d758e4f6fd1c)

Reply:
It was decided that SWITCH module is not needed and the only module in smartswitch is DPU.